concourse/steps/update_component_deps.py

```diff
@@ -503,7 +639,12 @@ def create_upgrade_pr(
             '\n\nRelease notes were shortened since they exceeded the maximum length allowed for a '
             'pull request body. The remaining release notes will be added as comments to this PR.'
         )
-        if max_pr_body_length < len(release_notes):
+
+        formatted_diff_length = len(f'## BoM Diff:\n{formatted_diff}\n\n')
```

Comment:
did you also consider that BoM Diff could exceed max PR body size?

Reply:
good catch! we should determine whether max-body-size would be exceeded, and in that case resort to just adding a link in that case